src/main/java/org/jabref/gui/copyfiles/CopyFilesDialogViewModel.java

```diff
@@ -0,0 +1,28 @@
+package org.jabref.gui.copyfiles;
+
+import java.util.List;
+
+import javafx.beans.property.SimpleListProperty;
+import javafx.collections.FXCollections;
+
+import org.jabref.gui.AbstractViewModel;
+
+public class CopyFilesDialogViewModel extends AbstractViewModel {
+
+    private final SimpleListProperty<CopyFilesResultItemViewModel> copyFilesResultItems = new SimpleListProperty<>(
+            FXCollections.observableArrayList());
+
+    //TODO: How do I add the data here?
+    public CopyFilesDialogViewModel(List<CopyFilesResultItemViewModel> items) {
```

Comment:
How do I get the data here? Should I add a field/property to the state Manager?
This is still one of the things which freaks me out of our javafx concept and for which I can't get my head around

Reply:
Yes, this is a bit ugly.

In the view you create an "injection context"
https://github.com/JabRef/jabref/blob/455965e931ca347e5caf91f52ded8e42cf11881c/src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FileAnnotationTabView.java#L13-L22
which then injects your data directly into the controller by
https://github.com/JabRef/jabref/blob/455965e931ca347e5caf91f52ded8e42cf11881c/src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FileAnnotationTabController.java#L43-L46

Reply:
But that does only work if I have a single View and not a dialog, right?
I still find this complicated and I don't see how I can achieve this with my custom dialog. I think I will add a property to the stage manager

Reply:
I really would appreciate, if you could hook into this  PR and add the things

Reply:
I think, as AbstractDialog derives from AbstractView, you just need to add a new constructor to the former one that relays everything to the view.

Sorry, I have enough "Baustellen" at the moment.